sklearn/decomposition/online_lda.py

```diff
@@ -277,13 +280,18 @@ def __init__(self, n_topics=10, doc_topic_prior=None,
         self.n_jobs = n_jobs
         self.verbose = verbose
         self.random_state = random_state
+        if n_components == 10 and n_topics != 10:
```

Comment:
Our convention is to do this with fit so it also applies when set_params is used